### PR TITLE
Add safe-swap app

### PR DESCRIPTION
**Pull request by anxolin**
Adds a new "safe-swap" App.
Part of https://github.com/gnosis/gp-ui/issues/273

This PR includes a new app using a very similar structure as the Explorer. For example, it also has the "network in url" thing.

For now, only has a simple page: `SwapPage`.
Adds basic state, navigation, network detection. Also uses the same kind of link to block explorer as the Explorer.

## Not included
Other things coming, in next PRs:
* Automatic link creation in PRaul.
* SDK integration
* Readme instructions

## Test
```bash
yarn start:safe-swap
```

image

### CODE REVIEW

File: src/apps/safe-swap/SafeSwapApp.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    import(
+      /* webpackChunkName: "Trade_chunk"*/
+      './pages/Swap'
+    ),
+)
+
+/**
+ * Update the global state
+ */
+export function StateUpdaters(): JSX.Element {
```
Comment (alfetopito) [inline src/apps/safe-swap/SafeSwapApp.tsx:48]: Minor nitpick: I don't think this needs to be exported

File: src/apps/safe-swap/SafeSwapApp.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+/** Redirects to the canonnical URL for mainnet */
+const RedirectMainnet = (): JSX.Element => {
+  const { pathname } = useLocation()
+
+  const pathMatchArray = pathname.match('/mainnet(.*)')
+  const newPath = pathMatchArray && pathMatchArray.length > 0 ? pathMatchArray[1] : '/'
+
+  return <Redirect push={false} to={newPath} />
+}
```
Comment (alfetopito) [inline src/apps/safe-swap/SafeSwapApp.tsx:87]: Can be potentially refactored into an utils?

File: src/apps/safe-swap/components/common/BlockExplorerLink/index.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { BlockExplorerLinkNetworkState } from 'components/common/BlockExplorerLink'
+
+/**
+ * Explorer app version of BlockExplorerLink
```
Comment (alfetopito) [inline src/apps/safe-swap/components/common/BlockExplorerLink/index.tsx:4]:
```suggestion
 * Safe app version of BlockExplorerLink
```

**Review by alfetopito — APPROVED** [review]
Looks good, only minor comments and nitpicks

File: webpack.config.js [hunk 4]
```diff
@@ -16,7 +16,8 @@ const CONFIG = loadConfig()
 const config = overrideEnvConfig(CONFIG)
 const TRADE_APP = { name: 'trade', title: 'Gnosis Protocol Exchange', filename: 'trade.html' }
 const EXPLORER_APP = { name: 'explorer', title: 'Gnosis Protocol Explorer', filename: 'index.html' }
-const ALL_APPS = [TRADE_APP, EXPLORER_APP]
+const SAFE_SWAP_APP = { name: 'safe-swap', title: 'Gnosis Safe Swap', filename: 'index.html' }
```
Comment (alfetopito) [inline webpack.config.js:19]: The title feels like this is something from the Safe.
What about: 

```suggestion
const SAFE_SWAP_APP = { name: 'safe-swap', title: 'Gnosis Swap - Safe app', filename: 'index.html' }
```

File: src/apps/safe-swap/SafeSwapApp.tsx (hunk 1 above)
Comment (anxolin) [reply src/apps/safe-swap/SafeSwapApp.tsx:48]: u right. I also removed the export in the explorer

File: src/apps/safe-swap/SafeSwapApp.tsx (hunk 2 above)
Comment (anxolin) [reply src/apps/safe-swap/SafeSwapApp.tsx:87]: Very good idea. I've moved it next to updater, so it's shared and close to the related logic

**Review by W3stside — COMMENTED** [review]
approved

**Pull request merged**